### ISSUE TITLE
Implementation challenges in tabular Data problems

### ISSUE THREAD

**Issue by arunizta** [muratsensoy/muratsensoy.github.io#2]
I was trying to implement this algorithm for a classification problem involving two classes. The final class probabilities after training seemed to be concentrated in [0.49, 0.52], where variations in the probability across classes and across observations are varying in 3/4 decimal points. 

Some high level details of the parameter setting that I used: 
1) evidence type : exponential evidence 
2) loss function : Bayes risk with cross entropy loss
3) num epochs :50 
5) num raw features : 99 , ( each transformed into embedding vector of dimension 16 )
6) batch size : 4096
7) Annealing Coeff : min(1.0, t/10) ∈ [0, 1] is the annealing coefficient, t is the index of the current training epoch, 
8) Event rate in the data is 45%.

Would like to understand from you, what may cause the predicted probability to lie in such a concentrated region. 
Have you seen the model performance being retained even after incorporation of the uncertainty based loss functions across different learning tasks (computer vision, text, tabular data)
